programs/library-bridge/SharedLibraryHandler.cpp

```diff
@@ -0,0 +1,233 @@
+#include "SharedLibraryHandler.h"
+
+#include <boost/algorithm/string/split.hpp>
+#include <ext/scope_guard.h>
+#include <IO/ReadHelpers.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int EXTERNAL_LIBRARY_ERROR;
+    extern const int SIZES_OF_COLUMNS_DOESNT_MATCH;
+}
+
+
+SharedLibraryHandler::SharedLibraryHandler(
+    const std::string & library_path_,
+    const std::string & library_settings)
+    : library_path(library_path_)
+{
+    library = std::make_shared<SharedLibrary>(library_path, RTLD_LAZY
```

Comment:
Seems like we have the same error as before here? We `dlopen` library for each new dictionary even if it uses an already opened library?